freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+| `offset`       | No       | Skips the given number of first rows from being returned. Defaults to not skip any rows if not provided.                                                       |
+| `order_by`     | No       | Sorts the row based on the given column values in either ascending or descending order. Defaults to not sort the rows.                                         |
+
+We will utilise the [GViz API][GVizAPI] to return the matching rows from the spreadsheet by running the Google Query below:
+
+`SELECT <column-1>, <column-2>, ..., <column-n> WHERE A IS NOT NULL [AND <condition>] [ORDER BY <ordering-1>, <ordering-2>, ..., <ordering-n>] [LIMIT <limit>] [OFFSET <offset>]`
+
+Note: Query inside square bracket denotes an optional query, only need to be provided if necessary.
+
+- Replace the `<column-1>, <column-2>, ..., <column-n>` based on the columns that are specified in the `columns` parameter.
+- If the client provided the `conditions` parameter, replace  `<condition>` with the actual client's condition, e.g.
+  suppose the `conditions` is `"B > 5"` then the `WHERE` clause will be equal to `WHERE A IS NOT NULL AND B > 5`.
+- If the client provided `limit` or `offset` parameter,  replace `<limit>` and `<offset>` with the given value respectively.
+- If the client provided `order_by` parameter, replace the `<ordering-1>, <ordering-2>, ..., <ordering-n>` with the
+  value given by the client, e.g. suppose the `order_by` parameter is `A ASC, B DESC` then the `ORDER BY` clause will
+  be equal to `ORDER BY A ASC, B DESC`.
+
+On how to use GViz API can refer to the appendix below.
+
+#### Update
+
+Update selected rows with the given value.
+
+The operation method should expect these parameters:
+| Parameter Name | Required | Remarks                                                                                                                                                                |
+| -------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `value`        | Yes      | the value that we're going to replace the rows with.                                                                                                                   |
+| `conditions`   | No       | Condition that must be true in order for the row to be updated in [Google Sheet Query Language][QueryLanguage] format. Defaults to update all columns if not provided. |
+
+To get the list of affected rows, we need to call GViz API with the following query:
+`SELECT A WHERE A IS NOT NULL [AND <condition>]`
+
+Note: `A` column is referring to the `_rid` column (which contains the row's index).
+
+After we get the list of indices that we need to update, call [spreadsheets.values.batchUpdate][BatchUpdateAPI] to
+update all the affected rows with the given `value`.
```

Comment:
This part is unclear on how the `value` parameter is being used in this API actually. Should it replace all the values? Or just the affected cells?

Reply:
Updated